src/cloud/components/RateLimitAlert.tsx

```diff
@@ -45,51 +62,74 @@ const RateLimitAlert: FC<Props> = ({
   alertOnly,
   className,
   resources,
+  showUpgrade,
+  sendNotify,
+  handleShowOverlay,
+  handleDismissOverlay,
 }) => {
+  const appearOverlay = () => {
+    handleShowOverlay('write-limit', null, handleDismissOverlay)
+  }
+
+  useEffect(() => {
+    if (
+      CLOUD &&
+      status === LimitStatus.EXCEEDED &&
+      resources.includes('write')
+    ) {
+      if (showUpgrade) {
+        sendNotify(
+          writeLimitReached(
+            '',
+            <UpgradeContent
+              type="write"
+              link="https://docs.influxdata.com/influxdb/v2.0/write-data/best-practices/optimize-writes/"
+              className="flex-upgrade-content"
+            />
+          )
+        )
+      } else {
+        sendNotify(
+          writeLimitReached(
+            "Data in has stopped because you've hit the query write limit. Let's get it flowing again: ",
+            <Button
+              className="rate-alert-overlay-button"
+              color={ComponentColor.Primary}
+              size={ComponentSize.Small}
+              onClick={appearOverlay}
+              text="Request Write Limit Increase"
+            />
+          )
+        )
+      }
+    }
+  }, [showUpgrade, status])
```

Comment:
should we listen to changes in the resources?

Reply:
Listening to changes on the resources is unnecessary, it updates anyways by virtue of the referential inequality (i think), ive tested locally and it seems to update correctly